src/main/java/uk/gov/hmcts/ccd/domain/types/DynamicListValidator.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.ccd.domain.types;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseField;
+
+@Named
+@Singleton
+public class DynamicListValidator implements BaseTypeValidator {
+    protected static final String TYPE_ID = "DynamicList";
+    private static final String DYNAMIC_LIST_ITEMS = "dynamic_list_items";
+    private static final String CODE = "code";
+    private static final String DEFAULT = "default";
+    private static final String LABEL = "label";
+
+    @Override
+    public BaseType getType() {
+        return BaseType.get(TYPE_ID);
+    }
+
+    @Override
+    public List<ValidationResult> validate(String dataFieldId, JsonNode dataValue, CaseField caseFieldDefinition) {
```

Comment:
beware if you look at FixedListValidator there are some validation checks there that we don't have here. For example https://github.com/hmcts/ccd-data-store-api/blob/5f9adffe7b15c8e81273a275f1bf6def3d6a76ca/src/main/java/uk/gov/hmcts/ccd/domain/types/FixedListValidator.java#L35 
but others too?

Reply:
Spoke to @amansinghb and there would be a separate ticket dealing with validations comming soon